api-tests/api.test.js

```diff
@@ -40,6 +40,38 @@ expect.extend({
     }
     return { pass: true }
   },
+  toBeAMatch(received) {
+    try {
+      expect(received.key).toBeA(String)
+      expect(received.time).toBeADateString()
+      expect(received.redScore).toBeUndefinedOr(Number)
+      expect(received.blueScore).toBeUndefinedOr(Number)
+      expect(received.redAlliance).toEqual(expect.any(Array))
+      expect(received.redAlliance).toHaveLength(3)
+      received.redAlliance.forEach(team => {
+        expect(team).toBeATeamKey()
+      })
+      expect(received.blueAlliance).toEqual(expect.any(Array))
+      expect(received.blueAlliance).toHaveLength(3)
+      received.blueAlliance.forEach(team => {
+        expect(team).toBeATeamKey()
+      })
+      expect(Object.keys(received)).toBeASubsetOf([
+        'key',
+        'time',
+        'redAlliance',
+        'blueAlliance',
+        'redScore',
+        'blueScore',
+      ])
+    } catch (error) {
+      return {
+        message: () => `expected ${received} to be a match`,
```

Comment:
It seems like this won't return a very useful error message when it fails.. Would it be reasonable to split this into multiple try-catches with more specific errors? Or pass along the actual error as a part of the message?